build.assets/grpcbox.mk

```diff
@@ -14,9 +14,14 @@ else
 GRPCBOX ?= $(GRPCBOX_BASE_NAME):$(BUILDBOX_VERSION)
 endif
 
+# UID and GID are used to run the grpcbox as the current user.
+UID := $$(id -u)
+GID := $$(id -g)
+
 # GRPCBOX_RUN has the necessary invocation to run a command inside the grpcbox.
 # Use this variable to run it from other Makefiles.
-GRPCBOX_RUN := docker run -it --rm -v "$$(pwd)/../:/workdir" -w /workdir $(GRPCBOX)
+# Set XDG_CACHE_HOME to let non-root user cache dependencies: https://github.com/bufbuild/buf/blob/f38ba9455c6650947b0b710327f9acb708da1196/private/pkg/app/app_unix.go#LL70C18-L70C32
```

Comment:
Public docs source:

* https://buf.build/docs/ci-cd/setup/#ci-caching
* https://buf.build/docs/bsr/module/manage/#module-license.

Reply:
You mean https://buf.build/docs/bsr/module/manage/#module-cache ? Is the cache even relevant if we don't mount it?

Reply:
Yes, #module-cache. The same variables seem to apply - first link only mentions BUF_CACHE_DIR, 2nd mentions XDG_CACHE_HOME (used in the PR).

Reply:
Setting only `BUF_CACHE_DIR` is not enough. I'm still getting 
```
failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied
```